mlflow_databricks_artifacts/store/artifact_repo.py

```diff
@@ -0,0 +1,323 @@
+import base64
+import logging
+import os
+import posixpath
+import requests
+import uuid
+
+from mlflow import get_tracking_uri
+from mlflow.entities import FileInfo
+from mlflow.exceptions import MlflowException
+from mlflow.store import artifact_repo
+from mlflow.protos.service_pb2 import MlflowService, GetRun, ListArtifacts
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE, INTERNAL_ERROR
+
+try:
+    from mlflow.protos.databricks_artifacts_pb2 import (
+        DatabricksMlflowArtifactsService,
+        GetCredentialsForWrite,
+        GetCredentialsForRead,
+        ArtifactCredentialType,
+    )
+except Exception:
+    from mlflow_databricks_artifacts.protos.databricks_artifacts_pb2 import (
+        DatabricksMlflowArtifactsService,
+        GetCredentialsForWrite,
+        GetCredentialsForRead,
+        ArtifactCredentialType,
+    )
+
+from mlflow_databricks_artifacts.azure.client import put_block, put_block_list
+from mlflow_databricks_artifacts.utils.databricks_utils import get_databricks_host_creds
+from mlflow_databricks_artifacts.utils.file_utils import relative_path_to_artifact_path, yield_file_in_chunks
+from mlflow_databricks_artifacts.utils.proto_json_utils import message_to_json
+from mlflow_databricks_artifacts.utils.rest_utils import (
+    call_mlflow_endpoint,
+    extract_api_info_for_service,
+    http_put,
+    http_get,
+)
+from mlflow_databricks_artifacts.utils.uri import (
+    extract_and_normalize_path,
+    get_databricks_profile_uri_from_artifact_uri,
+    is_databricks_acled_artifacts_uri,
+    is_valid_dbfs_uri,
+    remove_databricks_profile_info_from_artifact_uri,
+)
```

Comment:
These utilities have been extracted out of MLflow, since they are internal APIs / are version-dependent and are therefore subject to change / breakage.